python/ray/tests/test_runtime_env_complicated.py

```diff
@@ -329,6 +319,46 @@ def f():
     assert ray.get(f.remote())
 
 
+def test_inject_ray_and_python():
+    num_tests = 4
+    conda_dicts = [None] * num_tests
+    outputs = [None] * num_tests
+
+    conda_dicts[0] = {}
+    outputs[0] = {
+        "dependencies": ["python=7.8", "pip", {
+            "pip": ["ray==1.2.3"]
+        }]
+    }
+
+    conda_dicts[1] = {"dependencies": ["blah"]}
+    outputs[1] = {
+        "dependencies": ["blah", "python=7.8", "pip", {
+            "pip": ["ray==1.2.3"]
+        }]
+    }
+
+    conda_dicts[2] = {"dependencies": ["blah", "pip"]}
+    outputs[2] = {
+        "dependencies": ["blah", "pip", "python=7.8", {
+            "pip": ["ray==1.2.3"]
+        }]
+    }
+
+    conda_dicts[3] = {"dependencies": ["blah", "pip", {"pip": ["some_pkg"]}]}
+    outputs[3] = {
+        "dependencies": [
+            "blah", "pip", {
+                "pip": ["some_pkg", "ray==1.2.3"]
+            }, "python=7.8"
+        ]
+    }
+
+    for i in range(num_tests):
+        assert (inject_ray_and_python(conda_dicts[i], "ray==1.2.3",
+                                      "7.8") == outputs[i])
```

Comment:
very much a nit but writing the test this way will make it somewhat harder to debug because the traceback will just give you a failure on this line and it may be hard to decipher which test case failed. At a minimum printing the test case index before the assert would help

Reply:
Thanks, I hadn't thought of that

Reply:
Yeah, or using `parameters` :'(

Reply:
Ah, you mean `pytest.mark.parametrize()`?  Or something else?

Reply:
Yep!